src/snowflake/cli/plugins/spcs/image_registry/commands.py

```diff
@@ -39,3 +41,18 @@ def url(**options) -> MessageResult:
         raise ClickException(
             "No image repository found. To get the registry url, please switch to a role with read access to at least one image repository or create a new image repository first."
         )
+
+
+@app.command()
+@with_output
+@global_options_with_connection
+def login(**options) -> MessageResult:
+    """Logs in to the account image registry with the current user's credentials. Must be called from a role that can view at least one image repository in the image registry."""
+    try:
+        return MessageResult(RegistryManager().docker_registry_login().strip())
+    except NoImageRepositoriesFoundError:
```

Comment:
Maybe instead of catch and throw new error, extend ClickException by NoImageRepositoriesFoundError and add error message?

Reply:
I see your point, especially as that would reduce some of the code needed. 

My thought process was that we should have a separation of concerns here such that the manager class deals with the "backend" logic and throws errors related to that (data not found, not right format, can't query, etc.) which the commands definition can then catch and use to generate an informative error message to display on the frontend via ClickException.

Do you think that directly throwing a ClickException (or a subclass of ClickException) would be more efficient here?

Reply:
In my opinion using subclass of ClickException is a better solution. Keeps everything in one place. In your approach, you raise an error, catch somewhere higher and reraise as other error. If you have few `NoImageRepositoriesFoundError` how could you know which one cause error? If you add more types of errors, list of excepts is required here.

Reply:
Using ClikcException is already commonly used in our code base